### PR TITLE
feat(dashboard): Native filters - add type to native filter configuration

### PR DESCRIPTION
**Pull request by m-ajay**
### SUMMARY
In a future PR, we  will add a "Section" component to the native filters. We will need a new key to distinguish between them.

### TESTING INSTRUCTIONS
**Testing existing filters**
1. Run migration using `superset db upgrade`
2. Verify in the dashboards DB table that all the filters have a type (Check `json_metadata` column in dashboards table)
**Testing new filters** 
1. Add a new filter
2. Make sure that the type key appears in the new filter (Check `json_metadata` column in dashboards table)

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by graceguo-supercat** [comment]
Hi @m-ajay this PR is good. so when this PR is merged and migration is executed, native filter component should always has a `type` property, but how about new filter component in the new dashboard, which will be created in the near future? 
Besides migration, should we also add default `type` in the code?

**Comment by m-ajay** [comment]
> Hi @m-ajay this PR is good. After this PR is merged and migration is executed, all exist native filter components should have a `type` property, but how about new filter component in the new dashboard, which will be created in the near future?
> Besides migration, should we also add default `type` in the code at the same time?

@grace Thanks for the review.
I'm adding `type` as `NATIVE_FILTER` when creating a new filter. In the future, when we add a section, we will add `type` as `SECTION`

File: superset/migrations/versions/021b81fe4fbb_add_type_to_native_filter_configuration.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+            logger.info(
+                "[AddTypeToNativeFilter] Skipping Dashboard<pk:%s> ", dashboard.id
+            )
+            continue
+
+        json_meta = json.loads(dashboard.json_metadata)
+
+        for native_filter in json_meta.get("native_filter_configuration", []):
+            native_filter.setdefault("type", "NATIVE_FILTER")
+        dashboard.json_metadata = json.dumps(json_meta)
```
Comment (john-bodley) [inline superset/migrations/versions/021b81fe4fbb_add_type_to_native_filter_configuration.py:67]: For efficiency we likely should only be updating `dashboard.json_metadata` for those dashboards which were modified, so maybe something of the form, 

```python
if "native_filter_configuration" in json_meta:
    for native_filter in json_meta["native_filter_configuration"]:
        native_filter["type"] = "NATIVE_FILTER"
    
    dashboard.json_metadata = json.dumps(json_meta)
```

**Review by etr2460 — APPROVED** [review]
this lgtm, i love the verbose logging in the db migration too, it makes it easy to understand exactly what's happening

**Pull request merged**